src/Dataset.ts

```diff
@@ -577,6 +577,101 @@ export default class Dataset {
     }
     return this._allPrefixes;
   }
+
+  private async apiNDEDatasetRegister(method: "validate" | "submit", next?: (shaclReport: n3.Store) => void) {
+    const info = await this.getInfo();
+    if (info.accessLevel !== "public") {
+      throw getErr(
+        `Only datasets with accesslevel 'public' can be submitted your dataset has accesslevel '${info.accessLevel}'.`
+      );
+    }
+    const { consoleUrl } = await this._app.getInfo();
+    const apiUrl =
+      "https://datasetregister.netwerkdigitaalerfgoed.nl/api/datasets" + (method === "validate" ? "/validate" : "");
+    const datasetURL = `${consoleUrl}/${info.owner.accountName}/${info.name}`;
+    const data = { "@id": datasetURL };
+    const init: RequestInit = {
+      method: method === "submit" ? "POST" : "PUT",
+      headers: {
+        "Content-Type": "application/ld+json",
+        Accept: "text/turtle",
+        Link: '<http://www.w3.org/ns/ldp#RDFSource>; rel="type",<http://www.w3.org/ns/ldp#Resource>; rel="type"',
+      },
+      body: JSON.stringify(data),
+    };
+    await fetch(apiUrl, init).then(async (rs) => {
+      const rs2store = async (): Promise<n3.Store> => {
+        const store = new n3.Store();
+        const parser = new n3.Parser();
+        parser.parse(await rs.text(), (error, quad) => {
+          if (error) {
+            throw getErr(
+              "Failed to load response from NDE Dataset Register API as valid Turtle. Please contact NDE about this issue."
+            );
+          } else if (quad) {
+            store.addQuad(quad);
+          }
+        });
+        return store;
+      };
+      if (!rs.ok) {
+        if (next && rs.status === 400) {
+          next(await rs2store());
```

Comment:
an alternative to this is always throwing an error, but adding the shacl report in the `cause` of the error. E.g. `throw new Error("some message", {cause: await rs2store()})`
This will show up nicely in stack traces, and anybody catching this error can get the report from the error

Reply:
The reason for this pattern is that this submitting to NDE is build for use in TriplyETL. I do not want to throw an error, but receive the SHACL report as Store, so I might upload it to TriplyDB or generate a nice looking SHACL validation report.